docs/Functional Documentation/Functional_Specification.md

```diff
@@ -143,21 +144,171 @@ The codebase will be written in JavaScript as per decision made in [ADR-001](htt
 
 **Methods:**
 
-1. Get Dictionary
+1. `Get Dictionary`
    - Input: acts on Block
    - Output: Dictionary
    - Purpose: constructs and returns a dictionary containing the necessary data from the block: a list of hashes of transactions (transaction_hashes), the hash of the previous block (hash_previous_block), and the nonce (nonce).
-2. Get hash
+   - Error handling: None   
+
+2. `Get hash`
    - Input: acts on Block
    - Output: hashed JSON data
    - Purpose: This method calculates the current block's hash by first converting the block's data into a dictionary using the Get Dictionary method, then serializing that dictionary to a JSON-formatted string, and finally hashing the JSON string using the imported hash function. The resulting hash is the one associated with the current block.
+   - Error handling: None
 
 ##### 1.2.b Generation of the Genesis block with an initial value of Hash, Transaction, Nonce
 
 <br>
 
 #### 1.3 Transactions
 
+##### 1.3.a "UTXO" class
+
+This code defines a UTXO class representing a type of transaction output commonly used in blockchain systems. The class has methods to obtain the UTXO's attributes as a dictionary and to calculate a hash value based on these attributes.
+
+**Modules called:**
+
+1. JSON encoder module
+2. Module that provides a hash function:
+   - encode string to UTF-8
+   - hash object
+   - encode to base64
+   - decode to UTF-8
+
+**Attributes:**
+
+- Hash of transaction
+- Public key
+- Message
+
+**Methods:**
+
+1. `Get Dictionary`
+   - Input: acts on UTXO
+   - Output: Dictionary (originally: Python format)
+   - Purpose: This method returns a dictionary containing the UTXO's attributes (tx_hash, public_key, and message) as key-value pairs. The method creates the dictionary using the current instance's attribute values.
+   - Error handling: None
+
+2. `Get Hash`
+   - Input: acts on UTXO
+   - Output: hashed value of UTXO's dictionary
+   - Purpose: This method calculates a hash value for the dictionary representation of the UTXO. It calls the Get Dictionary (see 1. above) method to obtain the UTXO's attribute dictionary. It then converts the dictionary to a JSON string using json.dumps. The resulting JSON string is passed to the hashing.hash function (provided by the hashing module) to compute a hash value. The calculated hash value is returned as the result of the Get Hash method.
+   - Error handling: None

```

Comment:
hey @JCRigo86, sounds like we should have some error handling here and in other places as well

Reply:
hey @rkokhatskyi reviewing this spec, all methods here except one (Sign method in "unsigned transaction" class) actually act on "self" and not on inputs. I've updated the Error handling for that Sign method, however I'm not sure which error handling to set in the other methods. Can we consider it as solved until we have better visibility on how the code integrates overall? Unless you have suggestions on those error handlings

Reply:
thanks @JCRigo86, let's close this off therefore 👍